actionview/lib/action_view/helpers/asset_tag_helper.rb

```diff
@@ -364,6 +375,16 @@ def multiple_sources_tag_builder(type, sources)
           end
         end
 
+        def resolve_image_source(source, skip_pipeline)
+          if source.is_a?(Symbol) || source.is_a?(String)
```

Comment:
Oh, to be fair that is not a requirement indeed. But I think in this case it is better to check `to_model`. It will avoid the rescue of `NoMethodError` and it also would avoid checking the type of the arguments.